src/WooCommerce.php

```diff
@@ -1394,6 +1401,12 @@ public static function woocommerce_before_edit_address_form_shipping() {
    */
   public static function woocommerce_save_account_details_errors(\WP_Error $errors, $user) {
     $current_user = wp_get_current_user();
+
+    // Store a copy to conditionally display a message after form submission.
+    static::$userBeforeUpdate = clone $current_user;
+    static::$userBeforeUpdate->first_name = static::$userBeforeUpdate->first_name;
+    static::$userBeforeUpdate->last_name = static::$userBeforeUpdate->last_name;
```

Comment:
I don't get this lines. Are you setting those object properties to their same value? Why? Am I missing something?

Reply:
The code relies on PHP's loose typing and class property overloading:

Assigning a value to a non-existing property creates the property ad-hoc (unless the class implements the magic `__set()` method with code to explicitly prevent it).

As soon as the properties were created, the magic `__get()` method is no longer invoked upon access, because the stored value is immediately returned instead (like any other object property value).

Reply:
Thanks a lot for such detailed and didactic explanation.